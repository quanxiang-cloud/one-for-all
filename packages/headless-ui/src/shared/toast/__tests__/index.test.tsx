import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Icon from '@one-for-all/icon'

import { Notice } from '../index';

const toastPropsTest: ToastProps = {
  modifier: 'success',
  content: 'Test toast success!',
  duration: 1000,
  icon: <Icon name="close" size={18}/>,
  onClose: () => {
    console.log('Success toast is closed');
  },
}


test('tab navs props test', () => {
  const { container } = render( <Notice notice={{id:'1', ...toastPropsTest}} />);
  expect(container.querySelector('.ofa-notice')).toHaveTextContent('Test toast success!')
  expect(container).toMatchSnapshot();
})