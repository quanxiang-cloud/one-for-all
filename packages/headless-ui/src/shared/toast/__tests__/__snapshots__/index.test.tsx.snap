// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab navs props test 1`] = `
<div>
  <div
    class="ofa-notice ofa-notice__success ofa-notice-in"
    id="1"
  >
    <div
      class="ofa-notice-icon"
    >
      <svg
        class="ofa-svg-icon "
        data-name="close"
        style="width: 18px; height: 18px;"
      >
        <use
          xlink:href="/dist/images/sprite-137db0aa0124f9675ced9fff276d1044.svg#close"
        />
      </svg>
    </div>
    <div
      class="ofa-notice-content"
    >
      Test toast success!
    </div>
    <div
      class="ofa-notice-close-btn"
    >
      <svg
        class="ofa-svg-icon "
        data-name="close"
        style="width: 20px; height: 20px;"
      >
        <use
          xlink:href="/dist/images/sprite-137db0aa0124f9675ced9fff276d1044.svg#close"
        />
      </svg>
    </div>
  </div>
</div>
`;
